src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -897,12 +897,21 @@ private void AssertNoScriptTrees()
         /// Returns a new ArrayTypeSymbol representing an array type tied to the base types of the
         /// COR Library in this Compilation.
         /// </summary>
-        public IArrayTypeSymbol CreateArrayTypeSymbol(ITypeSymbol elementType, int rank = 1)
+        public IArrayTypeSymbol CreateArrayTypeSymbol(ITypeSymbol elementType, int rank = 1, NullableAnnotation elementNullableAnnotation = NullableAnnotation.NotApplicable)
         {
-            return CommonCreateArrayTypeSymbol(elementType, rank);
+            return CommonCreateArrayTypeSymbol(elementType, rank, elementNullableAnnotation);
         }
 
-        protected abstract IArrayTypeSymbol CommonCreateArrayTypeSymbol(ITypeSymbol elementType, int rank);
+        /// <summary>
+        /// Returns a new ArrayTypeSymbol representing an array type tied to the base types of the
+        /// COR Library in this Compilation.
+        /// </summary>
+        public IArrayTypeSymbol CreateArrayTypeSymbol(ITypeSymbol elementType, int rank)
```

Comment:
Please add a comment like we do for the CompilationOptions constructors saying that this method is for backcompat and to not touch the signature.

Reply:
And similar for the rest of the binary compat overloads.

---
In reply to: [302302333](https://github.com/dotnet/roslyn/pull/37048#discussion_r302302333) [](ancestors = 302302333)